includes/class-wc-payments-utils.php

```diff
@@ -110,7 +110,40 @@ public static function esc_interpolated_html( $string, $element_map ) {
 	 * @return int The amount in cents.
 	 */
 	public static function prepare_amount( $amount, $currency = 'USD' ) {
-		return round( (float) $amount * 100 );
+		$conversion_rate = 100;
+
+		if ( in_array( strtolower( $currency ), self::zero_decimal_currencies(), true ) ) {
+			$conversion_rate = 1;
+		}
+
+		return round( (float) $amount * $conversion_rate );
+	}
+
+	/**
+	 * List of currencies supported by Stripe, the amounts for which are already in the smallest unit.
+	 * Sourced directly from https://stripe.com/docs/currencies#zero-decimal
+	 *
+	 * @return array $currencies
+	 */
+	public static function zero_decimal_currencies() {
```

Comment:
Adapted directly from the Stripe extension function [here](https://github.com/woocommerce/woocommerce-gateway-stripe/blob/c6ca5552ead2b43c463fbdeecdea39179546b6b1/includes/class-wc-stripe-helper.php#L219-L244).